lib/workerclient.go

```diff
@@ -966,17 +995,23 @@ func (worker *WorkerClient) Write(ns *netstring.Netstring, nsCount uint16) error
 
 // setState updates the worker state
 func (worker *WorkerClient) setState(status HeraWorkerStatus) {
-	if worker.Status == status {
+	currentStatus := worker.Status
```

Comment:
Why is this explictly done..
What happens if worker.Status value changes..?
Shouldn't we use the reflected value in comparison?

Reply:
This is to avoid the race-condition on worker.Status value change. Currently we are not locking entire SetState operation. 
So there is possibility of worker.Status value can change between worker.Status == status check to Contains(validStateTransitionMap[worker.Status ], status). To avoid it we are using local variable.